edge/poll_async.go

```diff
@@ -261,30 +261,24 @@ func (service *PollService) processStackCommand(ctx context.Context, command cli
 
 	switch command.Operation {
 	case "add", "replace":
-		err = service.edgeStackManager.DeployStack(ctx, stackData)
-
-		if err != nil {
-			return service.portainerClient.SetEdgeStackStatus(stackData.ID, portainer.EdgeStackStatusError, stackData.RollbackTo, err.Error())
+		if err := service.edgeStackManager.DeployStack(ctx, stackData); err != nil {
+			return service.portainerClient.SetEdgeStackStatus(stackData.ID, portainer.EdgeStackStatusError, stackData.RollbackTo, "failed to deploy async stack. Error: "+err.Error())
```

Comment:
```suggestion
			return service.portainerClient.SetEdgeStackStatus(stackData.ID, portainer.EdgeStackStatusError, stackData.RollbackTo, fmt.Errorf("failed to deploy async stack: %w", err))
```

Reply:
I'd recommend using the fmt.Errorf pattern for error handling: https://www.notion.so/portainerwiki/Logging-guidelines-e41a3f3259b6450fb69a346030822a27?pvs=4#fa260beaf28e4a25ac586b04da7ef530